### PR TITLE
ext: hal: nxp: mcu: add device files for RT1024

### PR DESCRIPTION
**Pull request by Lucien-Zhao**
Add mcux 2.8.0 soc drivers and header files for mimxrt1024

Note: SDK date for RT1024 need to modify after RT1024 launch

Origin: NXP MCUXxpresso SDK 2.8.0
URL: mcuxpresso.nxp.com
Maintained-by: External

Signed-off-by: Lucien Zhao <lucien.zhao@nxp.com>

### CODE REVIEW

File: mcux/README [hunk 1]
```diff
@@ -1,4 +1,4 @@
-MCUXpresso SDK
+sMCUXpresso SDK
```
Comment (galak) [inline mcux/README:1]: remove change here.

File: mcux/README [hunk 2]
```diff
@@ -23,6 +23,7 @@ Status:
    MIMXRT1010	SDK 2.8.2 (2020-08-19)
    MIMXRT1015	SDK 2.8.2 (2020-08-19)
    MIMXRT1021	SDK 2.8.2 (2020-08-19)
+   MIMXRT1024	SDK 2.8.0 (2020-10-23)
```
Comment (mmahadevan108) [inline mcux/README:26]: Can we update RT1024 to SDK 2.9. I think this will be the latest at the time of release.

Comment (Lucien-Zhao) [reply mcux/README:26]: SDK 2.9 is version of RELEASE13 and RELEASE13 is in the first halt of next year. These codes is developed based on SDK2.8 and the last digit number 0 represents the number of times with SDK2.8.

Comment (mmahadevan108) [reply mcux/README:26]: Can you update to the latest SDK package for 1024 available inside our internal system

Comment (mmahadevan108) [reply mcux/README:26]: Now that the files are available in the public MCUXpresso site, can you please check we are using the latest available. I see some differences in `fsl_clock.*` files

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**